### PR TITLE
Fixed OAuth setup bug for sandboxes

### PR DESCRIPTION
**Pull request by clintonb**
ECOM-2180

Replaces #2314 

@feanil

### CODE REVIEW

**Comment by clintonb** [comment]
@feanil @jibsheet @jimabramson @rlucioni ready for final review

File: playbooks/edx-east/oauth_client_setup.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+  hosts: all
+  sudo: True
+  gather_facts: True
+  vars_files:
+    - "roles/common/defaults/main.yml"
+    - "roles/edxapp/defaults/main.yml"
+    - "roles/insights/defaults/main.yml"
+    - "roles/ecommerce/defaults/main.yml"
+  roles:
+    - oauth_client_setup
```
Comment (rlucioni) [inline playbooks/edx-east/oauth_client_setup.yml:11]: Does the ecommerce playbook need to be updated to use this role?

**Comment by rlucioni** [comment]
I don't understand why a separate playbook is required for OAuth2 client configuration. @clintonb for those unfamiliar, can you provide some context on why this was necessary?

File: playbooks/edx-east/oauth_client_setup.yml (hunk 1 above)
Comment (feanil) [inline playbooks/edx-east/oauth_client_setup.yml:11]: no.

**Comment by feanil** [comment]
@rlucioni this is required to be seperate because it needs to be run on every sandbox as the sandbox is being provisioned, not just when the base box is being built.  This is because of the callback URLs will change with each sandbox.  It needs to be a playbook because it needs to be run in ansible_provision.sh and needs access to vars from the other roles.

@clintonb :+1:

**Comment by rlucioni** [comment]
Thanks @feanil. :+1:

**Pull request merged**